Thoms' changes to data format section (#64)
DanNixon/sine2020_software_guidelines_paper: guidelines_for_collaborative_development_of_sustainable_data_treatment_software.tex

A good data format will have open source definitions and documentation, making it easy to understand and to implement file loaders for the software.
It is likely that a library will already exist for loading standardised file formats and should be used whenever possible.
Any implementation of a standardised data format needs to follow the standard and
the data format should only diverge from it if it is absolutely necessary.
If the divergence from the standard is required and has a clear benefit, then these changes should be suggested to be added to the standard.

> In the neutron and muon science community NeXus \cite{K_nnecke_2015} has become the default format  for raw data and in some cases also for processed data \cite{cansas}.

Reviewer: ```suggestion
In the neutron and muon science community NeXus \cite{K_nnecke_2015} has become the default format for raw data and processed data.
```

Although I'm not really sure we need to specify both raw and processed (what about simulated, that can also be stored in the NeXus format).

Author: I think it should be communities. Also it is not the default format for muons, only ISIS uses nexus. PSI uses bin and md, triumph has mud

Reviewer 2: I would like to promote the cansas format, but as far as I know SANS is the only community using nexus for processed data. There is an attempt to do something similar for reflectometry. McStas data can (and should) be stored in NeXus format.

Reviewer 2: I guess the point is that the community standards should be used wherever possible, so what about:

The Crystallographic Information File (CIF) is a good example of a community standard, in this case sponsored by the International Union of Crystallography \cite{cif}, which should be used for exchange of crystallographic data whenever possible.

For unprocessed neutron scattering data NeXus \cite{K_nnecke_2015} has become the default format and NeXus is also used for processed small angle neutron scattering data in the form of the canSAS format \cite{cansas}.

A few comments: 
* CIF can be both Crystallographic Information File and Crystallographic Information Framework. The later includes the former (https://www.iucr.org/resources/cif)
* After "whenever possible" we could add "including processed neutron diffraction data" but I am not sure to which extent this is done today. 
* unprocessed can of course also be raw, but maybe unprocessed is better understood by the readers.

NeXus aims to standardise the way data is represented by different facilities and software by defining a structure on top of the HDF5 file format \cite{HDF5}.
NeXus is a self describing file format where classes can be used to make it easy to understand the data, with no prior knowledge.

> Useful metadata (such as instrument configurations and data processing history) is trivial to store in NeXus  and is an important aspect for making data FAIR (i.e. Findable, Accessible, Inter-operable, and Reusable) \cite{panosc}.

Reviewer: ```suggestion
Useful metadata (such as instrument configurations and data processing history) is trivial to store in NeXus and is an important aspect for making data FAIR (i.e. Findable, Accessible, Inter-operable, and Reusable) \cite{panosc}.
```

I also feel FAIR needs more discussion.

Reviewer 2: Could we add:
Proper meta-data are required for being able to find the data through a search query in a data catalogue such as icat \cite{icat} or scicat \cite{scicat} or Google Dataset Search engine \cite{google_dataset_search}. Meta-data are also important for ensuring that the generation of the data set is fully understood to ensure they are used in the right context. This is crucial for the inter-operabiity and reusability of the data set. Examples of meta-data are instrument configurations, data processing history, sample environment setting, sample description, experiment proposal, etc. Standardized meta-data makes it possible to setup algorithms that automatically search and analyze a series of data set, for instance with machine learning techniques.

Then the former parenthesis with examples of meta data should be removed. 

Examples of machine learning techniques applied to data bases are NOMAD and the Organic Materials Database.

Author: ```suggestion
Useful metadata is trivial to store in NeXus and is an important aspect for making data FAIR (i.e. Findable, Accessible, Inter-operable, and Reusable) \cite{panosc}.
correctly labeled meta-data significantly improves the ability to use a search query in a data catalogue (e.g. icat \cite{icat}, scicat \cite{scicat} or Google Dataset Search engine \cite{google_dataset_search}). 
Well documented meta-data improves the reproducibility and reusability of the data.
Standardized metadata will improve how easy it is to identify key information.  
Examples of metadata are instrument configurations, data processing history, sample environment setting, sample description, experiment proposal etc. 
```
